assets/src/scripts/components/hydrograph/index.js

```diff
@@ -231,13 +229,22 @@ const attachToNode = function (node, {siteno} = {}) {
 
     let store = configureStore();
 
-    select(node)
+    store.dispatch(Actions.resizeTimeseriesPlot(node.offsetWidth));
+    let plotSelect = select(node);
+    plotSelect
         .call(provide(store))
-        .call(timeSeriesGraph)
         .select('.hydrograph-last-year-input')
             .on('change', dispatch(function () {
                 return Actions.toggleTimeseries('compare', this.checked);
             }));
+    plotSelect.append('div')
+            .attr('class', 'hydrograph-container')
+            .style('padding-bottom', ASPECT_RATIO_PERCENT)
+            .call(link(timeSeriesGraph, layoutSelector));
```

Comment:
Nesting calls to `link()` will be problematic, because the previously rendered nodes will still have their store subscriptions active.

We might eventually need a more general way to bind attribute values to nodes, but I think you can achieve the same thing with a call(link()) up on the svg node, like:

```
elem.append('svg')
    .call(link((elem, layout) => elem.attr('viewBox', `0 0 ${layout.width} ${layout.height}`), layoutSelector))
```

Reply:
I don't understand this: " the previously rendered nodes will still have their store subscriptions active". Can you explain? Does the state get messed up?

Reply:
The state will be okay, there will just be unneeded renders over the old, deleted DOM nodes.

The `connect` function registers a listener to Redux state changes and calls the provided function on state change. The `link` function wraps the connect function, to skip the call to the function if the selector it accepts returns the same object as last time.

When you do `.call(connect(..))` or `.call(link())`, in both cases the original D3 selector provided by `call` is saved, and passed in when the calls to actual rendering function is made. So what ends up happening, is those functions are called with the old D3 DOM selections, even though they're not in the document any more.

These higher order functions are hard to keep straight... I can clear up more on Slack if that doesn't make sense.

Reply:
I put in lots of breakpoints in and can see the extra render calls with the way I was doing it. Changing to just using the link on the code to add the viewBox makes way more sense. Thanks for explaining.

Reply:
💡 - I finally get it!

Reply:
> So what ends up happening, is those functions are called with the old D3 DOM selections, even though they're not in the document any more.

I think that mostly makes sense, though I'm confused by the above statement.... :thinking: